toptal/picasso
# chore(RichTextEditor): add links support to rich text editor

[COMM-1056]

### Description

Support for links need to be added into RichTextEditor & RichText components.

### How to test

- Access storybook.
- Play around with RichTextEditor in the `Links` story. Try adding links to selected text and replacing links by selecting an existing link and clicking the link button again. Try adding links without any selection and it should add it were the cursor is located at.
- CI should be 💚 

### Screenshots

![image](https://user-images.githubusercontent.com/9496960/174841158-8fc80231-d17f-494d-a11e-8461df2a706a.png)

### Development checks


**Breaking change**


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[COMM-1056]: https://toptal-core.atlassian.net/browse/COMM-1056?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
